src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -280,6 +285,119 @@ public void shutdown(ExecutorDriver d) {
     log.info("Executor asked to shutdown");
   }
 
+  protected void runHealthChecks(ExecutorDriver driver, Task task) {
+    log.info("Performing health check for task: " + task.taskInfo.getTaskId().getValue());
+    
+    String nodeName = null;
+    String healthCheckCmd = "netstat -plnat | grep ";
+    String taskIdStr = task.taskInfo.getTaskId().getValue();
+      
+    if (taskIdStr.contains(HDFSConstants.DATA_NODE_ID)) {
+      nodeName = HDFSConstants.DATA_NODE_ID;
+      healthCheckCmd = healthCheckCmd + HDFSConstants.DATA_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.JOURNAL_NODE_ID)) {
+      nodeName = HDFSConstants.JOURNAL_NODE_ID;
+      healthCheckCmd = healthCheckCmd + HDFSConstants.JOURNAL_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.ZKFC_NODE_ID)) {
+      nodeName = HDFSConstants.ZKFC_NODE_ID;
+      healthCheckCmd = healthCheckCmd + HDFSConstants.ZKFC_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.NAME_NODE_ID)) {
+      nodeName = HDFSConstants.NAME_NODE_ID;
+      healthCheckCmd = healthCheckCmd + HDFSConstants.NAME_NODE_PORT;
+    }
+      
+    try {
+      boolean nodeRegistered = false;
+      if (nodeName != null) {
+        nodeRegistered = runHealthCheckProcess(healthCheckCmd, nodeName);
+      }
+      if (!nodeRegistered) {
+        log.fatal("Node health check failed for task: " + task.taskInfo.getTaskId().getValue());
+        killTask(driver, task.taskInfo.getTaskId());
+        System.exit(2);
+      }
+    } catch (IOException | InterruptedException e) {
+      log.error("Error running health check: ", e);
+    }
+  }
+
+  private boolean runHealthCheckProcess(String healthCheckCmd, String nodeName) throws IOException,
+      InterruptedException {
+    String javaKeyWord = "java";
+    String psStr = "ps -efww";
+    boolean nodeRegistered = false;
+    Process healthCmd = Runtime.getRuntime().exec(new String[]{
+        "sh", "-c", healthCheckCmd});
+
+    if (healthCmd != null) {
+      int exitCode = healthCmd.waitFor();
+      if (exitCode != 0) {
+        log.error("Unable to run the health check command: " + healthCheckCmd + ", exit code: "
+            + exitCode);
+      } else {
+        InputStream inputStream = healthCmd.getInputStream();
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+          // Parse output to find process id
+          int endPortIndex = line.lastIndexOf(javaKeyWord);
+          int beginPortIndex = endPortIndex > 0 ? line.lastIndexOf(" ", endPortIndex) : -1;
+          if (endPortIndex != -1) {
+            line = line.substring(beginPortIndex + 1, endPortIndex - 1);
+            // Run ps command with the process id and check to see if it's output contains
+            // the node name
+            String psCmdStr = psStr + " " + line + " | grep " + nodeName;
```

Comment:
I think instead of doing this we should write a pid file when we start a task. Then we can just check if that pid is running. Something like `ps ax | grep -v grep | grep $pid`